pkg/apis/servicecatalog/types.go

```diff
@@ -270,9 +270,17 @@ type InstanceSpec struct {
 	// provisioned from.
 	PlanName string
 
-	// Parameters is a YAML representation of the properties to be
+	// List of sources to populate parameters.
+	// When a key exists in multiple
+	// sources, the value associated with the last source will take precedence.
+	// Values defined by a Parameter with a duplicate key will take precedence.
+	// +optional
+	ParametersFrom []ParametersFromSource
+
+	// Parameters is a set of the parameters to be
 	// passed to the underlying broker.
-	Parameters *runtime.RawExtension
+	// +optional
+	Parameters []Parameter
```

Comment:
Let's keep the design discussion in one place: https://github.com/kubernetes-incubator/service-catalog/pull/1075

@duglin I don't want to deprecate or drop `parameters` at this point. See the simplified design proposal.

I will update this PR according to the updated design.